src/core/utils/building/__tests__/detect-dev-changes.test.ts

```diff
@@ -184,16 +184,16 @@ describe('Detect Dev Changes', () => {
       const step1: BuildStepOutput = {
         entrypoints: [htmlPage1, htmlPage2],
         chunks: [
-          fakeOutputChunk({
-            moduleIds: [fakeFile(), changedPath],
+          fakeOutputAsset({
+            fileName: 'page1.html',
           }),
```

Comment:
This was a bad test. HTML files are assets, not chunks, so that's why this test was passing before. Making these changes before adding the fix reproduced the error.